astropy/units/tests/test_equivalencies.py

```diff
@@ -627,6 +628,58 @@ def test_spectraldensity_not_allowed(from_unit, to_unit):
         to_unit.to(from_unit, 1, u.spectral_density(1 * u.AA))
 
 
+@dataclasses.dataclass(eq=False)
+class DuckQuantity(np.lib.mixins.NDArrayOperatorsMixin):
+
+    quantity: u.Quantity
+
+    def __array_ufunc__(self, function, method, *inputs, **kwargs):
+
+        inputs = [
+            inp.quantity if isinstance(inp, type(self)) else inp for inp in inputs
+        ]
+
+        result = getattr(function, method)(*inputs, **kwargs)
+        if result is not NotImplemented:
+            return type(self)(result)
+        return NotImplemented
+
+    def to(self, unit: u.Unit, equivalencies: list[tuple] = []) -> "DuckQuantity":
+        return type(self)(self.quantity.to(unit, equivalencies))
+
+    def to_value(
+        self, unit: None | u.Unit = None, equivalencies: list[tuple] = []
+    ) -> "DuckQuantity":
+        return type(self)(self.quantity.to_value(unit, equivalencies))
```

Comment:
This is still inconsistent with how `Quantity` behaves: the `value` should not be a quantity duck type as well.

Thinking more about this, I wonder whether you could not make the test pass without any changes to astropy,
```
def __array__(self, dtype=None):
    return self.quantity.view(np.ndarray)
```

But obviously I'd need to know a bit more what the end goal is - if you want to base yourself on a different array class (or be agnostic to it), then it will not work.